pkg/kubelet/kubelet_pods.go

```diff
@@ -918,7 +918,8 @@ func (kl *Kubelet) PodResourcesAreReclaimed(pod *v1.Pod, status v1.PodStatus) bo
 		klog.V(3).Infof("Pod %q is terminated, but some containers have not been cleaned up: %s", format.Pod(pod), statusStr)
 		return false
 	}
-	if kl.podVolumesExist(pod.UID) && !kl.keepTerminatedPodVolumes {
+	checkVolumePaths := false
```

Comment:
@jingxu97 I believe we want to change this check to check for directory existence instead of mount point right?

Reply:
Hey @msau42 , Should I continue to this PR, is it still relevant?

Reply:
Hi @mucahitkurt yes this issue still exists. Fixing this will also fix this flaky test: https://github.com/kubernetes/kubernetes/issues/91001

Reply:
I added `podVolumePathsExist` check here.